monai/networks/utils.py

```diff
@@ -241,3 +243,36 @@ def pixelshuffle(x: torch.Tensor, dimensions: int, scale_factor: int) -> torch.T
     x = x.reshape(batch_size, org_channels, *([factor] * dim + input_size[2:]))
     x = x.permute(permute_indices).reshape(output_size)
     return x
+
+
+@contextmanager
+def eval_mode(*nets: nn.Module):
+    """
+    Set network(s) to eval mode and then return to original state at the end.
+
+    Args:
+        nets: Input network(s)
+
+    Examples
+
+    .. code-block:: python
+
+        t=torch.rand(1,1,16,16)
+        p=torch.nn.Conv2d(1,1,3)
+        print(p.training)  # True
+        with eval_mode(p):
+            print(p.training)  # False
+            print(p(t).sum().backward())  # will correctly raise an exception as gradients are calculated
+    """
+
+    # Get original state of network(s)
+    training = [n for n in nets if n.training]
+
+    try:
+        # set to eval mode
+        with torch.no_grad():
```

Comment:
perhaps add some checks for the attributes `hasattr(nets, "train")` `hasattr(nets, "training")`

Reply:
If we don't have the checks the exception will be raised and passed to the caller. Since this will happen on the first line I think it's safe to rely on Pythonic way of deferring checks to runtime. The requirement that the arguments are `nn.Module` instances should let users know what's expected as well.